### PR TITLE
OpenBSD 6.4 fixes

### PR DESCRIPTION
**Pull request by guedou**
This PR makes all unit tests working on OpenBSD 6.4.

### CODE REVIEW

File: scapy/arch/bpf/supersocket.py [hunk 1]
```diff
@@ -241,6 +241,8 @@ def get_frame(self):
         """Get a frame or packet from the received list"""
         if self.received_frames:
             return self.received_frames.pop(0)
+        else:
```
Comment (guedou) [inline scapy/arch/bpf/supersocket.py]: Avoid a race condition that could lead this function to return None.

File: scapy/utils.py [hunk 2]
```diff
@@ -1371,6 +1371,11 @@ def _write_header(self, pkt):
         if self.linktype is None:
             try:
                 self.linktype = conf.l2types[pkt.__class__]
+                # Import here to prevent import loops
+                from scapy.layers.inet import IP
+                from scapy.layers.inet6 import IPv6
+                if OPENBSD and isinstance(pkt, (IP, IPv6)):
```
Comment (guedou) [inline scapy/utils.py]: `tcpdump` on OpenBSD uses a specific DLT value for raw IP packets. This fixes some unit tests.

File: test/regression.uts [hunk 3]
```diff
@@ -6921,12 +6921,13 @@ assert r.linktype == DLT_EN10MB
...
+pcapfile = BytesIO(b'\xd4\xc3\xb2\xa1\x02\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\x00\x00\x01\x00\x00\x000}$]\xff\\\t\x006\x00\x00\x006\x00\x00\x00\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00E\x00\x00(\x00\x01\x00\x00@\x06|\xcd\x7f\x00\x00\x01\x7f\x00\x00\x01\x00\x14\x00P\x00\x00\x00\x00\x00\x00\x00\x00P\x02 \x00\x91|\x00\x000}$]\x87i\t\x00*\x00\x00\x00*\x00\x00\x00\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00E\x00\x00\x1c\x00\x01\x00\x00@\x11|\xce\x7f\x00\x00\x01\x7f\x00\x00\x01\x005\x005\x00\x08\x01r0}$]\xfbp\t\x00*\x00\x00\x00*\x00\x00\x00\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00E\x00\x00\x1c\x00\x01\x00\x00@\x01|\xde\x7f\x00\x00\x01\x7f\x00\x00\x01\x08\x00\xf7\xff\x00\x00\x00\x00')
```
Comment (guedou) [inline test/regression.uts]: This change, and below, ensure that the pcap and the tcpdump outputs are compatible with the OpenBSD implementation.

File: scapy/arch/bpf/supersocket.py (hunk 1 above)
Comment (p-l-) [reply scapy/arch/bpf/supersocket.py]: That's nothing but you don't need `else` after `return`.

**Pull request merged**